modules/graphdb/templates/07_cluster_setup.sh.tpl

```diff
@@ -127,45 +139,90 @@ if [ "$INSTANCE_ID" == "$${LOWEST_INSTANCE_ID}" ]; then
+  if [[ $IS_SECURITY_ENABLED == "true" ]]; then
+    echo "Security is enabled"
   else
-    echo "Failed setting GraphDB password. Please check the logs!"
-  fi
+    # Set the admin password
+    SET_PASSWORD=$(
+      curl --location -s -w "%%{http_code}" \
+        --request PATCH 'http://localhost:7200/rest/security/users/admin' \
+        --header 'Content-Type: application/json' \
+        --data "{ \"password\": \"$${GRAPHDB_PASSWORD}\" }"
+    )
+    if [[ "$SET_PASSWORD" == 200 ]]; then
+      echo "Set GraphDB password successfully"
+    else
+      echo "Failed setting GraphDB password. Please check the logs!"
+    fi
 
-  # Enable the security
-  ENABLED_SECURITY=$(curl -X POST -s -w "%%{http_code}" \
-    --header 'Content-Type: application/json' \
-    --header 'Accept: */*' \
-    -d 'true' 'http://localhost:7200/rest/security')
+    # Enable the security
+    ENABLED_SECURITY=$(curl -X POST -s -w "%%{http_code}" \
+      --header 'Content-Type: application/json' \
+      --header 'Accept: */*' \
+      -d 'true' 'http://localhost:7200/rest/security')
 
-  if [[ "$ENABLED_SECURITY" == 200 ]]; then
-    echo "Enabled GraphDB security successfully"
-  else
-    echo "Failed enabling GraphDB security. Please check the logs!"
+    if [[ "$ENABLED_SECURITY" == 200 ]]; then
+      echo "Enabled GraphDB security successfully"
+    else
+      echo "Failed enabling GraphDB security. Please check the logs!"
+    fi
   fi
+else
+  echo "The current instance: $INSTANCE_ID is not the lowest, skipping cluster creation"
+fi
+
+echo "####################################"
+echo "#    Updating GraphDB password     #"
+echo "####################################"
+
+if [[ -e "/var/opt/graphdb/password_creation_time" && "$INSTANCE_ID" == "$${LOWEST_INSTANCE_ID}" ]]; then
+    # The request will fail if the cluster state is unhealthy
+    # This handles rolling updates
+    for record in "$${ALL_DNS_RECORDS[@]}"; do
+      echo "Pinging $record"
+      # Removes the '.' at the end of the DNS address
+      cleanedAddress=$${record%?}
+      # Check if cleanedAddress is non-empty before calling check_gdb
+      if [ -n "$cleanedAddress" ]; then
+        while ! check_gdb "$cleanedAddress"; do
+          echo "Waiting for GDB $cleanedAddress to start"
+          sleep "$RETRY_DELAY"
+        done
+      else
+        echo "Error: cleanedAddress is empty."
+      fi
+    done
+
+    SET_NEW_PASSWORD=$(
+      curl --location -s -w "%%{http_code}" \
+        --request PATCH 'http://localhost:7200/rest/security/users/admin' \
+        --header 'Content-Type: application/json' \
+        --header 'Accept: text/plain' \
+        -u "admin:$${GRAPHDB_PASSWORD}" \
+        --data "{\"password\":\"$${GRAPHDB_ADMIN_PASSWORD}\",\"appSettings\":{\"DEFAULT_SAMEAS\":false,\"DEFAULT_INFERENCE\":false,\"EXECUTE_COUNT\":false,\"IGNORE_SHARED_QUERIES\":false,\"DEFAULT_VIS_GRAPH_SCHEMA\":false},\"grantedAuthorities\":[\"ROLE_USER\",\"READ_REPO_empty-rdfsplus\"]}"
```

Comment:
This would... be a problem. In order to avoid overriding potential updates in the user settings, you should first read them from the API, update the password and pass that as data.

Reply:
Now the script gets the config for the admin user before updating the password